src/sql-server-util/src/cdc.rs

```diff
@@ -310,60 +356,114 @@ pub enum CdcError {
 /// function.
 ///
 /// [`sys.fn_cdc_increment_lsn`](https://learn.microsoft.com/en-us/sql/relational-databases/system-functions/sys-fn-cdc-increment-lsn-transact-sql?view=sql-server-ver16)
-#[derive(Copy, Clone, Debug, Eq, PartialEq, Hash, Serialize, Deserialize, Arbitrary)]
-pub struct Lsn([u8; 10]);
+///
+/// Note: The derived impl of [`PartialOrd`] and [`Ord`] relies on the field
+/// ordering so do not change it.
+#[derive(
+    Default,
+    Copy,
+    Clone,
+    Debug,
+    Eq,
+    PartialEq,
+    PartialOrd,
+    Ord,
+    Hash,
+    Serialize,
+    Deserialize,
+    Arbitrary,
+)]
+pub struct Lsn {
+    /// Virtual Log File sequence number.
+    vlf_id: u32,
+    /// Log block number.
+    block_id: u32,
+    /// Log record number.
+    record_id: u16,
+}
 
 impl Lsn {
-    /// Interpret the provided bytes as an [`Lsn`].
-    pub fn interpret(bytes: [u8; 10]) -> Self {
-        Lsn(bytes)
-    }
+    const SIZE: usize = 10;
 
-    /// Return the underlying byte slice for this [`Lsn`].
-    pub fn as_bytes(&self) -> &[u8] {
-        self.0.as_slice()
-    }
+    /// Interpret the provided bytes as an [`Lsn`].
+    pub fn try_from_bytes(bytes: &[u8]) -> Result<Self, String> {
```

Comment:
If there are any links to documentation about the internal structure of this field including things like endianess it would be great to include them